src/mds/MDSRank.cc

```diff
@@ -3872,6 +3872,8 @@ const char** MDSRankDispatcher::get_tracked_conf_keys() const
     "mds_session_max_caps_throttle_ratio",
     "mds_symlink_recovery",
     "mds_session_metadata_threshold",
+    "mds_log_trim_threshold",
```

Comment:
89d6525b97b18fb893948578818a31be86833d93 has `runtime` flag set for `mds_log_trim_threshold` as well, is disallowing runtime modification for `mds_log_trim_threshold` intentional? does it pose any risk?

Reply:
I don't think it has any risk. We do allow runtime modification for `mds_log_trim_threshold`.

Reply:
so how about adding this as well
```
  if (changed.count("mds_log_trim_threshold")){
    log_trim_counter = DecayCounter(g_conf().get_val<double>("mds_log_trim_decay_rate"));
  }
```

Reply:
Not required since the udpated value is fetched in trim().

Reply:
okay but then the same can be done with `mds_log_trim_decay_rate` as well right? i understand that the decay rate needs to be modified with any changes on the fly and thus we have [this](https://github.com/ceph/ceph/pull/52652/files#diff-5400a459f43872ea2a7a5a9072105b6884c9058b9822d2897d79c7ff8ea54015R1627-R1629) but at every `trim()`, the way we get the updated value for `mds_log_trim_threshold`, we can also get the same for `mds_log_trim_decay_rate`. Is `DecayCounter` the reason behind keeping it in `MDLog::handle_conf_change`?

Reply:
Caching the config on update is preferred if the config is used in hot-code-paths and if the config is subjected to be changed by the admin. For this change, `mds_log_trim_threshold` isn't a config that admins would tune frequently and thereby is fetched only when needed.

That reminds me, we should add a note in the doc about how to tune these configs.

Reply:
yeah and needs a test case as well